include/ddc/kernels/splines/matrix.hpp

```diff
@@ -89,12 +74,14 @@ class Matrix
         return bx;
     }
 
-    template <class... Args>
-    Kokkos::View<double**, Args...> solve_batch_inplace(
-            Kokkos::View<double**, Args...> const bx) const
+    virtual ddc::DSpan2D_stride solve_transpose_inplace(ddc::DSpan2D_stride const bx) const
     {
         assert(int(bx.extent(0)) == m_n);
-        int const info = solve_inplace_method(bx.data(), 'N', bx.extent(1));
+        int const info = solve_inplace_method(
+                bx.data_handle(),
+                'T',
+                bx.extent(1),
+                std::max(bx.stride(0), bx.stride(1)));
```

Comment:
The value of the parameter `stride` looks suspicious to me. Shouldn't it be only `bx.stride(0)` ?

Reply:
The thing is `ddc::DSpan2D_stride` does not suppose a direction for the layout. In the current implementation I use a right layout for ginkgo backend (matrix_sparse) and a left layout for lapack backend (all the rest). It is not correctly asserted currently.

What do you think of changing the arguments of `solve_inplace_method` to stop using a raw pointer `double* b` and `size `and `stride` in favor of passing directly the `ddc::DSpan2D_stride` object ? I think it would simplify several things and make assertions easier to write

Reply:
Does it mean Ginkgo and lapack do not agree on which dimension should be contiguous ?

Reply:
`Arrays are passed as pointers, not as a pointer to pointers. All the LAPACKE routines that take one or more 2D arrays as a pointer receive a single extra parameter of type int. This argument must be equal to either LAPACK_ROW_MAJOR or LAPACK_COL_MAJOR which are defined in lapacke.h, specifying whether the arrays are stored in row-major or column-major order. If a routine has multiple array inputs, they must all use the same ordering.

Note that using row-major ordering may require more memory and time than column-major ordering, because the routine must transpose the row-major order to the column-major order required by the underlying LAPACK routine.`

Ok so the choice looks available in LAPACKE, not LAPACK, so we really don't want to support RightLayout with Lapack